test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -397,6 +397,157 @@ main() {
+            }
+           
+            @Props()
+            class FooProps extends UiProps {}
+           
+            @State()
+            class _\$FooState extends UiState {}
+           
+            @Component()
+            class FooComponent {
+              render() => null;
+            }
+          '''
+          );
+
+          var transformedSource = transformedFile.getTransformedText();
+          expect(transformedSource, contains(originalPrivateFooStateLine));
+          expect(transformedSource, isNot(contains(originalPublicFooStateLine)));
+          expect(transformedSource, contains(transformedFooStateLine));
+          expect(transformedSource, contains(fooStateImplExtendsPublicClass));
+          expect(transformedSource, isNot(contains(fooStateImplExtendsPrivateClass)));
+        });
+
+        test('with builder-compatible dual-class abstract props setup', () {
+          final originalPrivateClassLine = 'class _\$AbstractFooProps {';
+          final originalPublicClassLine = 'class AbstractFooProps extends _\$AbstractFooProps with _\$AbstractFooPropsAccessorsMixin {';
+          final transformedFooStateLine = 'class AbstractFooProps extends _\$AbstractFooProps';
+
+          preservedLineNumbersTest('''
+            @Factory()
+            UiFactory<FooProps> Foo;
+            
+            @Props()
+            class FooProps extends UiProps {}
+            
+            @Component()
+            class FooComponent {
+              render() => null;
+            }
+          
+            class AbstractFooProps extends _\$AbstractFooProps with _\$AbstractFooPropsAccessorsMixin {
+              // ignore: undefined_identifier, undefined_class, const_initialized_with_non_constant_value
+              static const AbstractFooPropsMeta meta = \$metaForAbstractFooProps;
+            }
+            
+            @AbstractProps() 
+            class _\$AbstractFooProps {
+              var bar;
+              var baz;
+            }
+          '''
+          );
+
+          var transformedSource = transformedFile.getTransformedText();
+          expect(transformedSource, contains(originalPrivateClassLine));
+          expect(transformedSource, isNot(contains(originalPublicClassLine)));
+          expect(transformedSource, contains(transformedFooStateLine));
+        });
+
+        test('with builder-compatible dual-class abstract state setup', () {
+          final originalPrivateClassLine = 'class _\$AbstractStateProps {';
+          final originalPublicClassLine = 'class AbstractStateProps extends _\$AbstractStateProps with _\$AbstractStatePropsAccessorsMixin {';
+          final transformedFooStateLine = 'class AbstractStateProps extends _\$AbstractStateProps';
+
+          preservedLineNumbersTest('''
+            @Factory()
+            UiFactory<FooProps> Foo;
+            
+            @Props()
+            class FooProps extends UiProps {}
+            
+            @Component()
+            class FooComponent {
```

Comment:
Should be able to ditch the factory, props, and component for this test since we only care about the abstract state